### PR TITLE
feat(proxy-cache): add wildcard and parameter match support for content_type

### PR DESCRIPTION
**Pull request by vm-001**
### Description

What did you change and why?
 
Include any supporting resources, e.g. link to a Jira ticket, GH issue, FTI, Slack, Aha, etc.

### Testing instructions

Netlify link: 

### Checklist 

- [x] Review label added 
- [x] PR pointed to correct branch (`main` for immediate publishing, or a release branch: e.g. `release/gateway-3.2`, `release/deck-1.17`)

[FTI-1131]

[FTI-1131]: https://konghq.atlassian.net/browse/FTI-1131?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/_hub/kong-inc/proxy-cache-advanced/_index.md [hunk 1]
```diff
@@ -57,7 +57,9 @@ params:
         - application/json
       datatype: array of string elements
       description: |
-        Upstream response content types considered cacheable. The plugin performs an **exact match** against each specified value; for example, if the upstream is expected to respond with a `application/json; charset=utf-8` content-type, the plugin configuration must contain said value or a `Bypass` cache status is returned.
+        Upstream response content types considered cacheable.
+        The wildcard content type can be used to match any specific type or subtype.
+        For example, `application/*` matches any subtypes of the application, `*/*` matches all content types. Note that `*/*` does not match with a content type that contains a parameter. If the response content type does not match the configuration, a `Bypass` cache status is returned.
```
Comment (mheap) [inline app/_hub/kong-inc/proxy-cache-advanced/_index.md:62]: @vm-001 Can I provide `application/json;*` to match `application/json; charset=utf-8` _and_ `application/json`?

Comment (vm-001) [reply app/_hub/kong-inc/proxy-cache-advanced/_index.md:62]: No, you can't. `application/json;*`  isn't a valid content type.

Comment (mheap) [reply app/_hub/kong-inc/proxy-cache-advanced/_index.md:62]: How about `application/json; charset=*`?

Our matching logic requires an exact match for the header. It'd be nice to say `application/json`, whether `charset=utf-8` is provided or not

Comment (vm-001) [reply app/_hub/kong-inc/proxy-cache-advanced/_index.md:62]: The matching behavior of Proxy Cache plugin now follows the Request Validator. I suggest let's keep the consistency with other plugins. Then, we can figure out the best match behavior and apply it to all plugins at once at the next stage. Do you agree with me?

**Comment by acgoldsmith** [comment]
@vm-001 , can you please advise the related KAG ticket? Thanks.

**Comment by vm-001** [comment]
@acgoldsmith On the PR description. FTI-1131

**Comment by hbagdi** [comment]
@mheap Could you review this once again?

File: app/_hub/kong-inc/proxy-cache-advanced/_index.md (hunk 1 above)
Comment (mheap) [inline app/_hub/kong-inc/proxy-cache-advanced/_index.md:62]:
```suggestion
        For example, `application/*` matches any subtypes of the application, `*/*` matches all content types. Note that `*/*` does not match with a content type that contains a parameter (e.g. `*/*` would not match `application/json; charset=utf-8`). If the response content type does not match the configuration, a `Bypass` cache status is returned.
```

File: app/_hub/kong-inc/proxy-cache/_index.md [hunk 2]
```diff
@@ -58,7 +58,9 @@ params:
         - application/json
       datatype: array of string elements
       description: |
-        Upstream response content types considered cacheable. The plugin performs an **exact match** against each specified value; for example, if the upstream is expected to respond with a `application/json; charset=utf-8` content-type, the plugin configuration must contain said value or a `Bypass` cache status is returned.
+        Upstream response content types considered cacheable. 
+        The wildcard content type can be used to match any specific type or subtype.
+        For example, `application/*` matches any subtypes of the application, `*/*` matches all content types. Note that `*/*` does not match with a content type that contains a parameter. If the response content type does not match the configuration, a `Bypass` cache status is returned.
```
Comment (mheap) [inline app/_hub/kong-inc/proxy-cache/_index.md:63]:
```suggestion
        For example, `application/*` matches any subtypes of the application, `*/*` matches all content types. Note that `*/*` does not match with a content type that contains a parameter (e.g. `*/*` would not match `application/json; charset=utf-8`). If the response content type does not match the configuration, a `Bypass` cache status is returned.
```

**Review by lena-larionova — APPROVED** [review]
LGTM

**Pull request merged**